edlib/src/edlib.tpp

```diff
@@ -1530,6 +1516,34 @@ std::unordered_map<Element, AlphabetIdx> transformSequences(
     return elementToAlphabetIdx;
 }
 
+/**
+ * Takes additional equalities for alphabet elements, transforms them into index equalities
+ * @param [in] additionalEqualities
+ * @param [in] additionalEqualitiesLength
+ * @param [in] elementToAlphabetIdx
+ * @return  transformedEqualities A vector of pairs in which each pair contains the indexes of an additional equality
+ */
+template<class Element, class AlphabetIdx>
+std::vector<EdlibEqualityPair<AlphabetIdx>> transformEqualityPairs(
+        const EdlibEqualityPair<Element>* additionalEqualities,
+        int additionalEqualitiesLength,
+        const std::unordered_map<Element,AlphabetIdx>& elementToAlphabetIdx){
+    std::vector<EdlibEqualityPair<AlphabetIdx>> transformedEqualities;
+    for(int i = 0; i < additionalEqualitiesLength; i++){
+        Element a = additionalEqualities[i].first;
+        Element b = additionalEqualities[i].second;
+        if (elementToAlphabetIdx.count(a) && elementToAlphabetIdx.count(b)) {
```

Comment:
If correct arguments are given, this check should not fail, right?

What I am worried about is if this check fails, we will never know -> it will be a silent fail!

If it fails, it means we made a mistake in our program, it is not users fault, in this case.
So the whole thing should crash. How about we put assert here, instead of this if-else?

Reply:
The reason I did this was that when at least one of the elements in an equality is not present in the alphabet, we can remove that equality. I thought that maybe the purpose of  `test12` in `runTests.cpp` is to ignore such equalities. Could you check that test? Maybe I have to change that test too?

Reply:
Ufff you are right! I am being silly, yes let's keep it this way!